[2025-07-11T11:33:40.070+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: kafka_to_postgres_dag.consume_and_store manual__2025-07-11T11:33:38.442308+00:00 [queued]>
[2025-07-11T11:33:40.080+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: kafka_to_postgres_dag.consume_and_store manual__2025-07-11T11:33:38.442308+00:00 [queued]>
[2025-07-11T11:33:40.081+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 1
[2025-07-11T11:33:40.099+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): consume_and_store> on 2025-07-11 11:33:38.442308+00:00
[2025-07-11T11:33:40.112+0000] {standard_task_runner.py:57} INFO - Started process 172 to run task
[2025-07-11T11:33:40.118+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'kafka_to_postgres_dag', 'consume_and_store', 'manual__2025-07-11T11:33:38.442308+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/kafka_to_postgres_dag.py', '--cfg-path', '/tmp/tmp5fi9ee83']
[2025-07-11T11:33:40.121+0000] {standard_task_runner.py:85} INFO - Job 3: Subtask consume_and_store
[2025-07-11T11:33:40.233+0000] {task_command.py:410} INFO - Running <TaskInstance: kafka_to_postgres_dag.consume_and_store manual__2025-07-11T11:33:38.442308+00:00 [running]> on host 2d5a9df6af8f
[2025-07-11T11:33:40.353+0000] {taskinstance.py:1545} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='kafka_to_postgres_dag' AIRFLOW_CTX_TASK_ID='consume_and_store' AIRFLOW_CTX_EXECUTION_DATE='2025-07-11T11:33:38.442308+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-11T11:33:38.442308+00:00'
[2025-07-11T11:33:40.354+0000] {kafka_to_postgres_dag.py:19} INFO - Connecting to Kafka...
[2025-07-11T11:33:40.360+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.5', 9092)]>: connecting to kafka:9092 [('172.18.0.5', 9092) IPv4]
[2025-07-11T11:33:40.365+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.5', 9092)]>: Broker version identified as 2.6
[2025-07-11T11:33:40.366+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.5', 9092)]>: Connection complete.
[2025-07-11T11:33:40.368+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('airflow-topic',)
[2025-07-11T11:33:40.369+0000] {kafka_to_postgres_dag.py:29} INFO - ‚úÖ Connected to Kafka
[2025-07-11T11:33:40.373+0000] {kafka_to_postgres_dag.py:35} INFO - ‚úÖ Connected to Postgres
[2025-07-11T11:33:40.377+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.5', 9092)]>: connecting to kafka:9092 [('172.18.0.5', 9092) IPv4]
[2025-07-11T11:33:40.377+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.5', 9092)]>: Connection complete.
[2025-07-11T11:33:40.377+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.5', 9092)]>: Closing connection. 
[2025-07-11T11:33:40.482+0000] {cluster.py:401} INFO - Coordinator for group/airflow-pg-group-new-0 is ('coordinator-1', 'kafka', 9092, None)
[2025-07-11T11:33:40.482+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group airflow-pg-group-new-0
[2025-07-11T11:33:40.483+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-11T11:33:40.484+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group airflow-pg-group-new-0
[2025-07-11T11:33:40.485+0000] {base.py:419} INFO - Failed to join group airflow-pg-group-new-0: NodeNotReadyError: coordinator-1
[2025-07-11T11:33:40.487+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.5', 9092)]>: connecting to kafka:9092 [('172.18.0.5', 9092) IPv4]
[2025-07-11T11:33:40.488+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.5', 9092)]>: Connection complete.
[2025-07-11T11:33:40.589+0000] {base.py:532} INFO - (Re-)joining group airflow-pg-group-new-0
[2025-07-11T11:33:40.592+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-6ef4aaae-31c0-40cd-9ac2-2801a0220eb7 for group airflow-pg-group-new-0; will retry join-group
[2025-07-11T11:33:40.592+0000] {base.py:419} INFO - Failed to join group airflow-pg-group-new-0: [Error 79] MemberIdRequiredError
[2025-07-11T11:33:40.593+0000] {base.py:532} INFO - (Re-)joining group airflow-pg-group-new-0
[2025-07-11T11:33:43.597+0000] {base.py:594} INFO - Successfully joined group airflow-pg-group-new-0 <Generation 3 (member_id: kafka-python-2.2.15-6ef4aaae-31c0-40cd-9ac2-2801a0220eb7, protocol: range)>
[2025-07-11T11:33:43.597+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-11T11:33:43.618+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('airflow-topic', 0)]
[2025-07-11T11:33:43.619+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('airflow-topic', 0),) for group airflow-pg-group-new-0
[2025-07-11T11:33:43.646+0000] {fetcher.py:418} INFO - Resetting offset for partition ('airflow-topic', 0) to offset 0.
[2025-07-11T11:33:43.748+0000] {kafka_to_postgres_dag.py:39} INFO - üì• Message: {"Player": "Ruturaj Gaikwad", "COUNTRY": "IND", "TEAM": "CSK", "AGE": 26, "CAPTAINCY EXP": 0, "Paying_Role": "Batting", "Mat": 36, "Inns": 36.0, "Runs": 1207.0, "BF": 926.0, "HS": 101.0, "Avg": "37.72", "SR": 130.35, "NO": 4.0, "4s": 113.0, "6s": 43.0, "0s": 4.0, "50s": 10.0, "100s": 1.0, "TMat": 9.0, "TInns": 8.0, "TRuns": 135.0, "TBF": 109.0, "THS": 57.0, "TAvg": 16.88, "TSR": 123.85, "TNO": 0.0, "T4s": 13.0, "T6s": 5.0, "T0s": 0.0, "T50s": 1.0, "T100s": 0.0, "B_Inns": null, "B_Balls": null, "B_Runs": null, "B_Maidens": null, "B_Wkts": null, "B_Avg": null, "B_Econ": null, "B_SR": null, "B_4w": null, "B_5w": null, "B_TInns": null, "B_TBalls": null, "B_TRuns": null, "B_TMaidens": null, "B_TWkts": null, "B_TAvg": null, "B_TEcon": null, "B_TSR": null, "B_T4w": null, "B_T5w": null, "SOLD_PRICE": "6cr"}
[2025-07-11T11:33:43.749+0000] {kafka_to_postgres_dag.py:54} ERROR - ‚ùå DAG crashed hard: relation "ipl_stats" does not exist
LINE 1: INSERT INTO ipl_stats (message) VALUES ('{"Player": "Ruturaj...
                    ^

[2025-07-11T11:33:43.750+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/kafka_to_postgres_dag.py", line 44, in consume_and_insert
    cursor.execute("INSERT INTO ipl_stats (message) VALUES (%s)", (clean_msg,))
psycopg2.errors.UndefinedTable: relation "ipl_stats" does not exist
LINE 1: INSERT INTO ipl_stats (message) VALUES ('{"Player": "Ruturaj...
                    ^

[2025-07-11T11:33:43.765+0000] {taskinstance.py:1345} INFO - Marking task as FAILED. dag_id=kafka_to_postgres_dag, task_id=consume_and_store, execution_date=20250711T113338, start_date=20250711T113340, end_date=20250711T113343
[2025-07-11T11:33:43.783+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 3 for task consume_and_store (relation "ipl_stats" does not exist
LINE 1: INSERT INTO ipl_stats (message) VALUES ('{"Player": "Ruturaj...
                    ^
; 172)
[2025-07-11T11:33:43.803+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2025-07-11T11:33:43.823+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
